src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
```

Comment:
The `waitFor` function is correctly set up to return a promise. This is a good approach for handling asynchronous events.